src/reference/00-Getting-Started/07A-Task-Graph.md

```diff
@@ -0,0 +1,324 @@
+  [Ant]: http://ant.apache.org/
+  [Rake]: https://ruby.github.io/rake/
+
+Task graph
+----------
+
+This page explains `build.sbt` definition in more detail.
+It assumes you've read [.sbt build definition][Basic-Def] and
+[scopes][Scopes].
+
+### What's the point of the build.sbt DSL?
+
+As we saw before, a [build definition][Basic-Def] consists of subprojects
+with a map called `settings` describing the subproject.
+There's more to the story.
+
+What the `Setting` sequence encodes is tasks and the dependencies among them,
+similar to [Make][Make] (1976), [Ant][Ant] (2000), and [Rake][Rake] (2003).
+
+#### Intro to Make
+
+The basic Makefile syntax looks like the following:
+
+```
+target: dependencies
+[tab] system command1
+[tab] system command2
+```
+
+Given a target (the default target is named `all`),
+
+1. Make checks if the target's dependencies have been built, and builds any of the dependencies that hasn't been built yet.
+2. Make runs the system commands in order.
+
+Let's take a look at a `Makefile`:
+
+```
+CC=g++
+CFLAGS=-Wall
+
+all: hello
+
+hello: main.o hello.o
+    \$(CC) main.o hello.o -o hello
+
+%.o: %.cpp
+    \$(CC) \$(CFLAGS) -c \$< -o \$@
+```
+
+Running `make`, it will by default pick the target named `all`.
+The target lists `hello` as its dependency, which hasn't been built yet, so Make will build `hello`.
+
+Next, Make checks if the `hello` target's dependencies have been built yet.
+`hello` lists two targets: `main.o` and `hello.o`.
+Once those targets are created using the last pattern matching rule,
+only then the system command is executed to link `main.o` and `hello.o` to `hello`.
+
+If you're just running `make`, you can focus on what you want as the target,
+and the exact timing and commands necessary to build the intermediate products are figured out by Make.
+We can think of this as dependency-oriented programming, or flow-based programming.
+Make is actually considered a hybrid system because while the DSL describes the task dependencies, the actions are delegated to system commands.
+
+#### Rake
+
+This hybridity is continued for Make successors such as Ant, Rake, and sbt.
+Take a look at the basic syntax for Rakefile:
+
+```ruby
+task name: [:prereq1, :prereq2] do |t|
+  # actions (may reference prereq as t.name etc)
+end
+```
+
+The breakthrough made with Rake was that it used a programming language to
+describe the actions instead of the system commands.
+
+#### Benefits of hybrid flow-based programming
+
+Rather than thinking `settings` as a key-value map,
+a better analogy would be to think of it as a DAG (directed acyclic graph)
+of tasks where the edges denote *happens-before*. Let's call this the _task graph_.
+There are several motivation to organizing the build this way.
+
+First is de-duplication. With flow-based programming, a task is executed only once even when it is depended by multiple tasks.
+For example, even when multiple tasks along the task graph depend on `compile in Compile`,
```

Comment:
again, this is the only reference to a scope in the entire chapter. For the sake of the comparison with Make (which AFAIK does not have one such distinction, but I'm no expert) I'd avoid talking about scopes here, and introduce them later, as an addendum.